lib/src/storage/backend/rocksdb.rs

```diff
@@ -619,16 +817,24 @@ pub struct Reader {
 enum InnerReader {
     Snapshot(Rc<InnerSnapshot>),
     Transaction(Weak<*mut rocksdb_transaction_t>),
+    Secondary(Arc<DbHandler>), // TODO: we should not hold the whole enum, because we have to use
+                               // secondary_db_handler_or_error to access rocksdb_t. we should
+                               // safe it directly?!
```

Comment:
Maybe a bad idea (and probably not for this PR), change the types to something like:
```rust
pub enum DB {
    ReadWrite(Arc<RwHandler>),
    ReadOnly(Arc<RoHandler>),
}

struct RwDbHandler { // Same as the previous struct
    db: *mut rocksdb_transactiondb_t,
    options: *mut rocksdb_options_t,
    transaction_options: *mut rocksdb_transaction_options_t,
    transactiondb_options: *mut rocksdb_transactiondb_options_t,
    read_options: *mut rocksdb_readoptions_t,
    write_options: *mut rocksdb_writeoptions_t,
    flush_options: *mut rocksdb_flushoptions_t,
    env_options: *mut rocksdb_envoptions_t,
    ingest_external_file_options: *mut rocksdb_ingestexternalfileoptions_t,
    compaction_options: *mut rocksdb_compactoptions_t,
    block_based_table_options: *mut rocksdb_block_based_table_options_t,
    column_family_names: Vec<&'static str>,
    cf_handles: Vec<*mut rocksdb_column_family_handle_t>,
    cf_options: Vec<*mut rocksdb_options_t>,
    path: PathBuf,
    in_memory: bool
}


struct RoDbHandler {
    db: *mut rocksdb_t,
    options: *mut rocksdb_options_t,
    read_options: *mut rocksdb_readoptions_t,
    env_options: *mut rocksdb_envoptions_t,
    column_family_names: Vec<&'static str>,
    cf_handles: Vec<*mut rocksdb_column_family_handle_t>,
    cf_options: Vec<*mut rocksdb_options_t>,
    path: PathBuf,
    secondary_path: Option<Path> // Maker of secondary vs read-only. TODO: this is not nice
}
```

Reply:
I don't fully get it. Maybe we can discuss this in an issue or new pull request?

Reply:
Yes, definitely!